pkg/apis/core/types.go

```diff
@@ -3718,6 +3718,15 @@ type ServiceSpec struct {
 	// This field is alpha-level and is only honored by servers that enable the ServiceLBNodePortControl feature.
 	// +optional
 	AllocateLoadBalancerNodePorts *bool
+
+	// loadBalancerClassName is the name of the load balancer implementation this Service belongs to.
```

Comment:
Interestingly in the KEP it referenced `loadBalancerClass` except in most place so agreed to drop `Name`. 

> Should clouds that currently implement custom annotations (e.g. networking.gke.io/load-balancer-type: "Internal") adopt this?

My thinking here is that the class name should be used to distinguish between controller implementations, so the annotation should still be used if the same controller uses the annotation to distinguish between "types". Also, it's just simpler to reason about this if we assume that any existing cloud-provider implementation requires an empty class name. While I do think it would be useful for cloud providers to claim certain class names, I think it would be overly complicated and not really worthwhile doing.

Reply:
> Should clouds that currently implement custom annotations (e.g. networking.gke.io/load-balancer-type: "Internal") adopt this?

On second thought, I'm inclined to say yes but for external cloud providers only. I'd like to address this in a follow-up PR because I'm not sure what it means yet for replacing existing annotations. Some cloud providers will have to support both and that's probably fine.

Reply:
I'd suggest then, that we change this to a label-style string like "company.com/something", it's unambiguous.  We can reserve non-prefixed names for end-user use.